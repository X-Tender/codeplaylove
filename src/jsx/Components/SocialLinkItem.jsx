import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
	url: PropTypes.string.isRequired,
	iconName: PropTypes.string.isRequired,
	children: PropTypes.string.isRequired,
};

const SocialLinkItem = ({ url, iconName, children }) => (
	<li className="social-links__list-item">
		<a
			className="social-links__link"
			href={url}
			name={children}
			rel="noopener noreferrer"
			target="_blank"
			title={children}
		>
			<i className={`fab fa-${iconName} fa-3x`} />
		</a>
	</li>
);

SocialLinkItem.propTypes = propTypes;

export default SocialLinkItem;
