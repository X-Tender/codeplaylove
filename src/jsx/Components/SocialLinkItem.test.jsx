import React from 'react';
import { shallow } from 'enzyme';
import SocialLinkItem from './SocialLinkItem';

describe('SocialLinkItem', () => {
	const url = 'https://github.com';
	const iconName = 'github';
	const children = 'gitHub';

	const comp = shallow(
		<SocialLinkItem iconName={iconName} url={url}>
			{children}
		</SocialLinkItem>
	);

	it('renders correctly', () => {
		expect(comp).toMatchSnapshot();
	});

	it('renders correct props', () => {
		expect(comp.find('li a').prop('href')).toEqual(url);
		expect(comp.find('li a').prop('name')).toEqual(children);
		expect(comp.find('li a').prop('title')).toEqual(children);
		expect(comp.find('li a i').hasClass(`fa-${iconName}`)).toBe(true);
	});
});
