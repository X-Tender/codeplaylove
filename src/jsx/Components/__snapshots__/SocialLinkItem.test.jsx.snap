// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialLinkItem renders correctly 1`] = `
<li
  className="social-links__list-item"
>
  <a
    className="social-links__link"
    href="https://github.com"
    name="gitHub"
    rel="noopener noreferrer"
    target="_blank"
    title="gitHub"
  >
    <i
      className="fab fa-github fa-3x"
    />
  </a>
</li>
`;
